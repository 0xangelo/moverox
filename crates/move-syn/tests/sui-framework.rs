use move_syn::{File, sanitize_for_tokenizer};
use testresult::TestResult;
use unsynn::{IParse as _, ToTokens as _};

macro_rules! test_file {
    ($($file: ident),* $(,)?) => {$(
        #[test]
        fn $file() -> TestResult {
            let content = include_str!(concat!(
                "../../../move/sui-framework/sources/",
                stringify!($file),
                ".move"
            ));
            let ast: File = sanitize_for_tokenizer(&content).into_token_iter().parse_all()?;
            insta::assert_snapshot!(ast.tokens_to_string());
            Ok(())
        }
    )*};
}

test_file!(
    accumulator,
    address,
    authenticator_state,
    bag,
    balance,
    bcs,
    borrow,
    clock,
    coin,
    config,
    deny_list,
    display,
    dynamic_field,
    dynamic_object_field,
    event,
    hex,
    linked_table,
    math,
    object,
    object_bag,
    object_table,
    package,
    party,
    pay,
    priority_queue,
    prover,
    random,
    sui,
    table,
    table_vec,
    token,
    transfer,
    tx_context,
    types,
    url,
    vec_map,
    vec_set,
    versioned,
);
