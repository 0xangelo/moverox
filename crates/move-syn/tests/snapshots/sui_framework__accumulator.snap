---
source: crates/move-syn/tests/sui-framework.rs
expression: ast.tokens_to_string()
---
module sui :: accumulator ; const ENotSystemAddress : u64 = 0 ; public struct AccumulatorRoot has key { id : UID , } # [allow (unused_function)] fun create (ctx : & TxContext) { assert ! (ctx . sender () == @ 0x0 , ENotSystemAddress) ; transfer :: share_object (AccumulatorRoot { id : object :: sui_accumulator_root_object_id () , }) }
